odoo/addons/base/tests/test_user_has_group.py

```diff
@@ -249,3 +248,41 @@ def test_implied_groups(self):
 
         with self.assertRaises(ValidationError):
             user_b.write({"groups_id": [(4, group_C.id)]})
+
+    def test_has_group_cleared_cache_on_write(self):
+        Users = self.env["res.users"]
+        self.env.registry._clear_cache()
+        self.assertEqual(
+            len(Users.pool._Registry__cache),
+            0,
+            "Ensure cache is empty (got: %r)" % self.env.cache._data['res.users']
+        )
```

Comment:
This can be simplified, since the registry is accessible from the test itself as `self.registry`.
```suggestion
        group0 = self.env.ref(self.group0)

        self.registry._clear_cache()
        self.assertFalse(self.registry._Registry__cache, "Ensure ormcache is empty")
```